### PR TITLE
Make Dataprep system test self-sufficient

### PR DESCRIPTION
**Pull request by moiseenkov**
1. Made Dataprep system test self-sufficient - from now it creates all needed objects and removes them afterwards.
2. For making the previous point achievable, I added a few methods into the hook and covered them by unit tests.
3. I also added a small bugfix for `DataprepGetJobsForJobGroupOperator` - it is now using connection id given by user instead of hardcoded value.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (eladkal) [inline airflow/providers/google/cloud/hooks/dataprep.py:147]: what is this link? is this google domain?
We must use only official links
I know links are by `https://cloud.google.com/dataprep`

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (eladkal) [inline airflow/providers/google/cloud/hooks/dataprep.py]: I don't know this API but please double check if this is what you want.
Changing this in the future means breaking change to the provider.
Some APIs are defaulted to latest version unless use specify specific version that might be better solution here if the API supports it.

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (eladkal) [inline airflow/providers/google/cloud/hooks/dataprep.py]: If we are left with v4 lets define it globally rather than in each function

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (eladkal) [inline airflow/providers/google/cloud/hooks/dataprep.py]: what is this link?

**Review by eladkal — DISMISSED** [review]

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/dataprep.py:147]: Dataprep by Trifacta is a service integrated with GCP but lives beyond it. That's why there's Trifacta's official documentation is used here, and Google's page `https://cloud.google.com/dataprep` points to Trifacta's resources.

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/dataprep.py]: I'm pretty confident that this is the needed API because it has already been used in the hook. I only added few more methods.

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/dataprep.py]: Good point, thanks! I added a parameter to the hook `api_version: str = "v4"`, so it's defined in a single place now.

File: airflow/providers/google/cloud/hooks/dataprep.py
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/dataprep.py]: Thanks for noticing. I aligned all links to the single source.

**Comment by moiseenkov** [comment]
@eladkal , hi
Could you please take a look at my fixes? Thank you!

**Comment by moiseenkov** [comment]
@potiuk , hi,
Could we merge it please?

**Comment by potiuk** [comment]
> @potiuk , hi, Could we merge it please?

Ass soon as @eladkal changes his review status. Until that I CAN't merge it.

**Review by potiuk — APPROVED** [review]

**Pull request merged**
